plugins/inputs/elasticsearch_query/aggregation_parser.go

```diff
@@ -0,0 +1,154 @@
+package elasticsearch_query
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	elastic "gopkg.in/olivere/elastic.v5"
+)
+
+type resultMetric struct {
+	name   string
+	fields map[string]interface{}
+	tags   map[string]string
+}
+
+func (e *ElasticsearchQuery) parseSimpleResult(measurement string, searchResult *elastic.SearchResult, acc telegraf.Accumulator) error {
+
+	fields := make(map[string]interface{})
+	tags := make(map[string]string)
+
+	fields["doc_count"] = searchResult.Hits.TotalHits
+
+	acc.AddFields(measurement, fields, tags)
+	return nil
+}
+
+func (e *ElasticsearchQuery) parseAggregationResult(aggregationQueryList []aggregationQueryData, searchResult *elastic.SearchResult, acc telegraf.Accumulator) error {
+	var measurements = map[string]map[string]string{}
+
+	// organize the aggregation query data by measurement
+	for _, aggregationQuery := range aggregationQueryList {
+
```

Comment:
Remove newline. Please only use newlines to indicate new "logical code blocks" as this clutters the code otherwise IMO. :-)